karapace/compatibility/__init__.py

```diff
@@ -48,14 +47,7 @@ def is_transitive(self) -> bool:
 
 
 def check_avro_compatibility(reader_schema, writer_schema) -> SchemaCompatibilityResult:
-    result = AvroChecker().get_compatibility(reader=reader_schema, writer=writer_schema)
-    if (
```

Comment:
TBH I never understood why this code existed. The correct behavior is what SR does.

Reply:
The same. I tried to understand the reasoning behind it but couldn't. It's basically just eating (valid) compatibility check errors. It could be a leftover of previous functionality or something like that.